imap_processing/spice/pointing_frame_handler.py

```diff
@@ -0,0 +1,242 @@
+"""
+Generate Pointing Frame.
+
+Notes
+-----
+Kernels that are required to run this code:
+1. imap_science_0001.tf - pointing frame kernel
+2. imap_sclk_0000.tsc - spacecraft clock kernel
+3. imap_wkcp.tf - spacecraft frame kernel
+4. naif0012.tls - standard NAIF leapsecond kernel
+5. imap_spin.bc - test attitude kernel generated by simulate_imap_attitude.m.
+These need to be placed in tests/pointing_frame/test_data.
+
+References
+----------
+https://spiceypy.readthedocs.io/en/main/documentation.html.
+"""
+
+import logging
+import os
+from pathlib import Path
+
+import numpy as np
+import spiceypy as spice
+from numpy.typing import NDArray
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+# TODO : Add multiple pointings to the pointing frame.
+
+
+def get_et_times(ck_kernel: str) -> tuple[float, float, np.ndarray]:
+    """
+    Create the pointing frame.
+
+    Parameters
+    ----------
+    ck_kernel : str
+        Path of ck_kernel used to create the pointing frame.
+
+    Returns
+    -------
+    et_start : float
+        Start time of ck_kernel.
+    et_end : float
+        End time of ck_kernel.
+    et_times : numpy.ndarray
+        Array of times between et_start and et_end.
+    """
+    # Get the spacecraft ID.
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.gipool
+    id_imap_spacecraft = spice.gipool("FRAME_IMAP_SPACECRAFT", 0, 1)
+
+    # TODO: Queried pointing start and stop times here.
```

Comment:
Can you please write a ticket for this with the SPICE label?

Reply:
https://github.com/IMAP-Science-Operations-Center/imap_processing/issues/739